src/main/java/reposense/system/CommandRunner.java

```diff
@@ -25,14 +25,37 @@ public static String gitLog(String root, Date fromDate, Date toDate) {
         return runCommand(rootPath, command);
     }
 
-    public static void checkOut(String root, String hash) {
+    public static void checkout(String root, String hash) {
         Path rootPath = Paths.get(root);
         runCommand(rootPath, "git checkout " + hash);
     }
 
-    public static String blameRaw(String root, String fileDirectory) {
+    /**
+     * Checkout to the latest commit before {@code untilDate} in {@code branchName} branch
+     * if {@code untilDate} is not null
+     */
+    public static void checkoutToDate(String root, String branchName, Date untilDate) {
```

Comment:
As quoted from git documentation
`When you are not interested in changes older than version v2.6.18, or changes older than 3 weeks, you can use revision range specifiers similar to git rev-list:`

if its similar to git rev list, is there any point in doing this?

Reply:
This is to ensure the `FileInfo` generated is based on the latest commit of the date range.
Should there be any changes to the files after we analyze from the latest commit of the date range, the blame results generated using this date range may not correspond to the correct lines of the 'updated' files, and we may even get `IndexOutOfBoundException` if the file has deleted lines.

Reply:
i see. can add it to the header comments too?